### ISSUE TITLE
Research whether a layout capture can be snapshotted as an artifact upon visibility failures

### ISSUE THREAD

**Issue by d4vidi** [wix/Detox#2078]
**Is your feature request related to a problem? Please describe.**
When tests fail due to visibility issues (95% of the time), at best, the user gets these artifacts hinting at the device's status at the time:
- Hierarchy log (sometimes truncated)
- Screenshot (sometime taken a bit too late)
From these, it's hard to understand what was *really* displayed on the device when the test failed.

**Describe the solution you'd like**
An additional, dynamically created, `xyz.li` file (i.e. readable by Android Studio's layout inspector), generated as a Detox artifact alongside all others.

**Describe alternatives you've considered**
Stick with what we have now :-/

**Additional context**
N/A

**Comment by LeoNatan** [comment]
On iOS, there isn't a similar file format which we can create. I once wanted to do something similar in Detox Instruments, but haven't worked on or thought about it in years.

Why isn't a screenshot and a hierarchy enough?

Another big problem with this approach is native hierarchy vs what web devs are familiar with, e.g. "components" and "props". There is no way to bring the latter, or somehow map it to the view hierarchy, leaving many people, unfamiliar with native development, scratching their heads. Those that are familiar usually have no problem following [this guide](https://github.com/wix/Detox/blob/master/docs/Guide.DebuggingInXcode.md) and checking the live hierarchy (even in 3D form) in Xcode.

**Comment by d4vidi** [comment]
I agree this may not be ideal for web devs. Nevertheless native devs could benefit, esp. when investigating issues on CI. Snapshots are too superficial, and the textual hierarchies are overly detailed.

**Comment by LeoNatan** [comment]
Apparently new Xcode 12 has something similar. I too will research the file format and try to implement for Detox.

**Comment by noomorph** [comment]
I wonder, can it be helpful: https://github.com/Grigory-Rylov/android-layout-inspector ?
